src/extension/wps/wps-core/src/main/java/org/geoserver/wps/gs/AggregateProcessPPIO.java

```diff
@@ -33,6 +37,10 @@ protected MapperWrapper wrapMapper(MapperWrapper next) {
             };
         };
         xstream.allowTypes(new Class[] { AggregateProcess.Results.class });
+        xstream.omitField(AggregateProcess.Results.class, "aggregateAttribute");
```

Comment:
So if someone uses the new group by functionality, the output will be available only in json?

Reply:
> I will put here some examples so we look both at the same output (my comments and questions at the end). Note, this are the outputs produced by this pull request.
> 
> #### WPS aggregate request WITHOUT group by attributes:
> **_XML_**
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <AggregationResults>
>     <Max>500.0</Max>
> </AggregationResults>
> ```
> 
> **_JSON_**
> 
> ```
> {
>   "AggregationFunctions": ["Max"],
>   "AggregationAttribute": "energy_consumption",
>   "GroupByAttributes": [],
>   "AggregationResults": [
>     [500]
>   ]
> }
> ```
> 
> #### WPS aggregate request WITH group by attributes:
> **_XML_**
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <AggregationResults>
>     <GroupByResult>
>         <object-array>
>             <string>FABRIC</string>
>             <big-decimal>500.0</big-decimal>
>         </object-array>
>         <object-array>
>             <string>HOUSE</string>
>             <big-decimal>6.0</big-decimal>
>         </object-array>
>         <object-array>
>             <string>SCHOOL</string>
>             <big-decimal>60.0</big-decimal>
>         </object-array>
>     </GroupByResult>
> </AggregationResults>
> ```
> 
> **_JSON_**
> 
> ```
> {
>   "AggregationFunctions": ["Max"],
>   "AggregationAttribute": "energy_consumption",
>   "GroupByAttributes": ["building_type"],
>   "AggregationResults": [
>     ["FABRIC",500],
>     ["HOUSE",6],
>     ["SCHOOL",60]
>   ]
> }
> ```
> 
> #### Comments
> * Answering your question: yes the current XML output can handle the group by result and is backwards compatible if there is no group by results (at least this was my intention).
> * Do you think is worth to add headers info to the XML output like in the JSON output (probably yes) ? The results are returned in the order of the invocations (group by results and functions this is also the case in the JSON format).
> * Do you think is worth add a documentation page for this process ? If yes where should I create a page in GeoServer documentation for this ?
> * Regarding the omit fields, I was only omitting fields I had to add to the process result so I had access to some extra information when producing the JSON output (to create the headers) since I don't produce headers in the XML output format I was just ignoring that information.
> * Looking at the examples if you don't agree with something please let me know.
> 
> The next time I will put better examples on my pull request description and thanks for the review.



> I will put here some examples so we look both at the same output (my comments and questions at the end). Note, this are the outputs produced by this pull request.
> 
> #### WPS aggregate request WITHOUT group by attributes:
> **_XML_**
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <AggregationResults>
>     <Max>500.0</Max>
> </AggregationResults>
> ```
> 
> **_JSON_**
> 
> ```
> {
>   "AggregationFunctions": ["Max"],
>   "AggregationAttribute": "energy_consumption",
>   "GroupByAttributes": [],
>   "AggregationResults": [
>     [500]
>   ]
> }
> ```
> 
> #### WPS aggregate request WITH group by attributes:
> **_XML_**
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <AggregationResults>
>     <GroupByResult>
>         <object-array>
>             <string>FABRIC</string>
>             <big-decimal>500.0</big-decimal>
>         </object-array>
>         <object-array>
>             <string>HOUSE</string>
>             <big-decimal>6.0</big-decimal>
>         </object-array>
>         <object-array>
>             <string>SCHOOL</string>
>             <big-decimal>60.0</big-decimal>
>         </object-array>
>     </GroupByResult>
> </AggregationResults>
> ```
> 
> **_JSON_**
> 
> ```
> {
>   "AggregationFunctions": ["Max"],
>   "AggregationAttribute": "energy_consumption",
>   "GroupByAttributes": ["building_type"],
>   "AggregationResults": [
>     ["FABRIC",500],
>     ["HOUSE",6],
>     ["SCHOOL",60]
>   ]
> }
> ```
> 
> #### Comments
> * Answering your question: yes the current XML output can handle the group by result and is backwards compatible if there is no group by results (at least this was my intention).
> * Do you think is worth to add headers info to the XML output like in the JSON output (probably yes) ? The results are returned in the order of the invocations (group by results and functions this is also the case in the JSON format).
> * Do you think is worth add a documentation page for this process ? If yes where should I create a page in GeoServer documentation for this ?
> * Regarding the omit fields, I was only omitting fields I had to add to the process result so I had access to some extra information when producing the JSON output (to create the headers) since I don't produce headers in the XML output format I was just ignoring that information.
> * Looking at the examples if you don't agree with something please let me know.
> 
> The next time I will put better examples on my pull request description and thanks for the review.

So, when WPS aggregate request WITH group by attributes,and output xml results, can I get the output in the order I entered it?

Reply:
and when i output json format, it doesn't seem to support Chinese

Reply:
@nmco

Reply:
![微信图片_20230309141331](https://user-images.githubusercontent.com/61198510/223942251-29ab87dc-e55a-4d77-ba19-5fed90b33e43.png)